### ISSUE TITLE
Train Model With Multiple Input Images

### ISSUE THREAD

**Issue by Neltherion** [mlfoundations/open_clip#435]
Is it possible to change the model to accept more than one image as the input?

If I'm not mistaken, CLIP takes an image and a text as the inputs, extracts the features of these two inputs and finally gives us the logits of the distance of the image to the text.

So, is it possible to give two (or more) input images and extract ONE feature from the input images (just like before)?

I want to somehow mix the two inputs. For example, inputting an image alongside it's semantic segmentation as the input to the model. If it's possible, what parts of the code should I change? Or is this already implemented and usable?

Thanks.

**Comment by rom1504** [comment]
There are multiple ways to do this
1. Late fusion: Use the existing model on all your images then aggregate
the embeddings (using a simple average or something stronger)
2. Early fusion: for this you would indeed need to adapt the openclip code,
then produce a large such (N image, text) dataset then retrain

I advise to start with 1

**Comment by Neltherion** [comment]
Thanks, I was really looking for approach #2😁 Is it possible and if it is, any hints on where to start?

**Comment by rom1504** [comment]
Maybe you could have a look at
https://github.com/LAION-AI/temporal-embedding-aggregation first
